### PR TITLE
Account linking: Display available financial providers [1/4]

### PR DESCRIPTION
**Pull request by kkzeng**
- Retrieve list of available financial providers from Firebase
- Display them using design-consistent ListTiles
- Add entry point in user dashboard

### CODE REVIEW

File: lib/ui/pages/account-linking/available_fsp.dart [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      final AvailableFSPController fspController = Get.find<AvailableFSPController>();
+      if(fspController.availableFsps.value.isEmpty) {
+        return _buildEmptyDisplay();
+      }
+
+      return ListView.builder(
+        itemCount: fspController.availableFsps.value.length + 2,
+        itemBuilder: (BuildContext ctxt, int index) {
+          switch(index) {
+            case 0:  return _buildIcon(); break;
```
Comment (looilee) [inline lib/ui/pages/account-linking/available_fsp.dart]: nit - extra white space

**Review by looilee — APPROVED** [review]

**Review by stevenwjy — COMMENTED** [review]
I feel like there are quite a number of formatting issues here... probably can use some auto format?

File: lib/controllers/ephemeral/account-linking/available_fsp_controller.dart [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+class AvailableFSPController extends GetxController {
+  AvailableFSPController(this._participantRepository);
+
+  final ParticipantRepository _participantRepository;
+  RxList<Fsp> availableFsps = <Fsp>[].obs;
+
+
+  @override
+  void onInit() {
+    availableFsps.bindStream(_participantRepository.listenForAvailableFSPs());
```
Comment (stevenwjy) [inline lib/controllers/ephemeral/account-linking/available_fsp_controller.dart]: I don't think it is necessary to bind the stream and keep listening here...
Why don't you just perform a get query upon opening the screen?

Comment (kkzeng) [reply lib/controllers/ephemeral/account-linking/available_fsp_controller.dart]: Yeah I guess that makes sense - I initially wrote it to just fetch once lol but then it seemed better at the time but I guess its unnecessary to listen

File: lib/repositories/firebase/participant_repository.dart [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import 'package:pispapp/models/fsp.dart';
+import 'package:pispapp/repositories/interfaces/i_participant_repository.dart';
+import 'package:cloud_firestore/cloud_firestore.dart' show Firestore;
+
+class ParticipantRepository implements IParticipantRepository {
+  final _colRef = Firestore.instance.collection('participants');
+
+  @override
+  Stream<List<Fsp>> listenForAvailableFSPs() {
```
Comment (stevenwjy) [inline lib/repositories/firebase/participant_repository.dart]: Is this function still needed?

**Review by stevenwjy — APPROVED** [review]
lgtm 👍

File: lib/repositories/firebase/participant_repository.dart (hunk 3 above)
Comment (kkzeng) [reply lib/repositories/firebase/participant_repository.dart]: Not really needed anymore. I can delete it before I merge

**Pull request merged**